tiledb/sm/filter/test/compile_all_filters_main.cc

```diff
@@ -29,7 +29,11 @@
 #include "../filter_create.h"
 
 int main() {
+  using namespace tiledb::sm;
+
   (void)sizeof(tiledb::sm::FilterCreate);
-  (void)&tiledb::sm::FilterCreate::deserialize;
+  (void)static_cast<shared_ptr<Filter> (*)(
```

Comment:
There are two functions in FilterCreate::deserialize and I think that both should be tested.

Reply:
There's no need. The voided-out references are used to get the linker to require a top-level symbol, and that brings everything in the compilation unit. The simpler-signature version of `FilterCreate::deserialize` uses the other one for its implementation and it's in the same compilation unit, so there's no need.

If linkers worked differently, such that each function were linked separately, you'd need to add a reference to every top-level function that _could_ be referenced. But as it stands you only need one.